readme.md
## pm2安装
```
npm install -g pm2
sudo ln -s /usr/local/nodejs/bin/pm2 /usr/local/bin
```

## 开启启动
```
pm2 save;
cat more ~/.pm2/dump.pm2
pm2 startup
```

## 启动命令
```
pm2 start npm --name changyou -- start
```

shell/history.txt
  1  ls
    2  cd /opt
    3  ls
    4  netstat -tunlp
    5  sudo docker ps
    6  top
    7  node -v
    8  sudo yum install gcc gcc-c++
    9  sudo yum install cmake*
   10  yum install zlib-devel bzip2-devel openssl-devel ncurses-devel sqlite-devel readline-devel tk-devel gdbm-devel db4-devel libpcap-devel xz-devel -y
   11  sudo yum install zlib-devel bzip2-devel openssl-devel ncurses-devel sqlite-devel readline-devel tk-devel gdbm-devel db4-devel libpcap-devel xz-devel -y
   12  python -V
   13  which python
   14  sudo yum install python-tools
   15  sudo yum install gtk2-devel
   16  ls
   17  cd /tmp
   18  ls
   19  tar -zxvf opencv-3.4.8.tar.gz
   20  cd opencv-3.4.8/
   21  mkdir build
   22  cd build
   23  ls
   24  sudo cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -D BUILD_opencv_python2=ON -D BUILD_opencv_python3=ON ..
   25  sudo yum install cmake*
   26  sudo cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -D BUILD_opencv_python2=ON -D BUILD_opencv_python3=ON ..
   27  sudo yum install cmake* --skip-broken
   28  sudo yum install cmake*
   29  rpm -Va --nofiles --nodigest
   30  sudo yum install cmake*
   31  yum install -y gcc gcc-c++ make automake 
   32  sudo yum install -y gcc gcc-c++ make automake 
   33  sudo yum install cmake*
   34  ls
   35  cd /tmp
   36  ls
   37  wget https://cmake.org/files/v3.12/cmake-3.12.0-rc1.tar.gz
   38  tar -zxvf cmake-3.12.0-rc1.tar.gz
   39  cd cmake-3.12.0-rc1
   40  ./bootstrap
   41  gmake
   42  gmake install
   43  sudo gmake install
   44  cmake --version
   45  cd /tmp
   46  ls
   47  cd opencv-3.4.8/
   48  ls
   49  cd build
   50  ls
   51  sudo cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -D BUILD_opencv_python2=ON -D BUILD_opencv_python3=ON ..
   52  cmake
   53  sudo cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -D BUILD_opencv_python2=ON -D BUILD_opencv_python3=ON ..
   54  cmake -D CMAKE_BUILD_TYPE=RELEASE -D CMAKE_INSTALL_PREFIX=/usr/local -D BUILD_opencv_python2=ON -D BUILD_opencv_python3=ON ..
   55  sudo make 
   56  sudo make
   57  sudo make install
   58  sudo yum -y install epel-release
   59  sudo yum -y install python-pip
   60  pip -V
   61  ls
   62  cd /tmp
   63  ls
   64  curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py
   65  python get-pip.py
   66  sudo yum -y install epel-release
   67  sudo yum -y install python-pip
   68  wget -q http://peak.telecommunity.com/dist/ez_setup.py
   69  python ez_setup.py
   70  ls
   71  sudo yum install python-setuptools
   72  easy_install pip
   73  sudo easy_install pip
   74  pip -V
   75  cd /opt
   76  ls
   77  sudo mkdir project
   78  ls
   79  cd project
   80  ls
   81  sudo mv /tmp/lee_slide_cracker.zip .
   82  ls
   83  sudo unzip lee_slide_cracker.zip 
   84  ls
   85  sudo rm -rf lee_slide_cracker.zip 
   86  ls
   87  cd lee_slide_cracker/
   88  ls
   89  sudo rm -rf venv/
   90  ls
   91  cd scripts/
   92  ls
   93  pip install -r requirements
   94  sudo pip install -r requirements
   95  sudo pip install Werkzeug
   96  sudo pip install subprocess32
   97  sudo yum install python-dev
   98  sudo yum install python-subprocess32
   99  sudo pip install -r requirements
  100  sudo pip uninstall subprocess32
  101  ls
  102  sudo vim requirements 
  103  sudo pip install -r requirements
  104  ls
  105  cd ..
  106  ls
  107  pwd
  108  ls
  109  nohup python /opt/project/lee_slide_cracker/run.py > /opt/project/lee_slide_cracker/run.log 2>&1 &
  110  sudo nohup python /opt/project/lee_slide_cracker/run.py > /opt/project/lee_slide_cracker/run.log 2>&1 &
  111  sudo chmod -R 777 /opt
  112  nohup python /opt/project/lee_slide_cracker/run.py > /opt/project/lee_slide_cracker/run.log 2>&1 &
  113  ls
  114  cat run.
  115  cat run.log 
  116  ps -ef|grep python
  117  ls
  118  pip install opencv-python
  119  sudo yum install libX*
  120  ls
  121  nohup python /opt/project/lee_slide_cracker/run.py > /opt/project/lee_slide_cracker/run.log 2>&1 &
  122  cat run.log 
  123  ps -ef|grep python
  124  ifconfig
  125  ls
  126  cd /opt
  127  ls
  128  sudo mv /tmp/nginx-1.17.7.tar.gz .
  129  ls
  130  tar -vxzf nginx-1.17.7.tar.gz 
  131  ls
  132  sudo rm -rf nginx-1.17.7.tar.gz 
  133  ls
  134  cd nginx-1.17.7/
  135  ls
  136  cd confi
  137  cd conf
  138  ls
  139  sudo vim nginx.conf
  140  nginx
  141  ls
  142  cd ..
  143  ls
  144  cd ..
  145  ls
  146  cd nginx-1.17.7/
  147  ls
  148  sudo yum install gcc gcc-c++ automake pcre pcre-devel zlip zlib-devel openssl openssl-devel 
  149  ./configure  --prefix=/usr/local/nginx  --sbin-path=/usr/local/nginx/sbin/nginx --conf-path=/usr/local/nginx/conf/nginx.conf --error-log-path=/var/log/nginx/error.log  --http-log-path=/var/log/nginx/access.log  --pid-path=/var/run/nginx/nginx.pid --lock-path=/var/lock/nginx.lock  --user=nginx --group=nginx --with-http_ssl_module --with-http_stub_status_module --with-http_gzip_static_module --http-client-body-temp-path=/var/tmp/nginx/client/ --http-proxy-temp-path=/var/tmp/nginx/proxy/ --http-fastcgi-temp-path=/var/tmp/nginx/fcgi/ --http-uwsgi-temp-path=/var/tmp/nginx/uwsgi --http-scgi-temp-path=/var/tmp/nginx/scgi --with-pcre
  150  make
  151  sudo make install
  152  nginx
  153  ls
  154  cd /usr/local
  155  ls
  156  cd nginx
  157  ls
  158  cd sbin
  159  ls
  160  nginx
  161  sudo nginx
  162  sudo ln -s /usr/local/nginx/sbin/nginx /usr/local/bin/
  163  nginx
  164  nginx -v
  165  ls
  166  cd ..
  167  ls
  168  cd /opt/projcec
  169  ls
  170  cd ..
  171  ls
  172  cd /opt
  173  ls
  174  mkdir logs
  175  ls
  176  cd logs
  177  ls
  178  mkdir lee_slide_cracker
  179  ls
  180  cd lee_slide_cracker/
  181  ls
  182  pwd
  183  ps -ef|grep pytho
  184  kill -9 23342
  185  ps -ef|grep pytho
  186  ps -ef|grep python
  187  kill -9 23351
  188  ps -ef|grep python
  189  ls
  190  ps -ef|grep python
  191  clear

config.js
const programName = 'changyou';
var config = {
	host: 'http://test-m-stg.ppppoints.com/butler/mbrauthorization.htm?',
	secret: '123456',
	imgAddress: 'http://172.27.83.80:8000/calpos',
	serverUrl: 'http://172.27.41.48:8070/',
	merId: 'S1000134',
	channelSource: '02000000',
	debug: true,
	plen: 1,	// 开几个浏览器处理绑定和兑换积分
	dlen: 1,	// 开几个容器处理授权
	smsTimes: 60, // 短信等待次数
	times: 10,	  // 滑动匹配最大次数
	checkchrom: 'Chromium --disable-background-networking',
	image: 'registry.cn-hangzhou.aliyuncs.com/terminal_project/pojie-changyou:1.17',
	top: 595,
	left: 110,
	width: 20,
	validWaitTimes: 6,
	redis:{
		port:6379,
		db: 7,
		host:'172.27.83.47',
		pass:'tryme'
	},
	mysql:{
		host: "172.27.83.63",
		port: 3306,
		user: "wulei",
		password: "tOvwycLUVAwq4ZlU",
		database: "se",
		connectionLimit: 10
	},
	log:{
		appenders: {
			console: { //记录器1:输出到控制台
				type: "console"
			},
			log_file: {	//记录器2：输出到文件
				type: "file",
				filename: __dirname + `/logs/${programName}.log`,
				maxLogSize: 10485760,
				layout: {
					type: "pattern",
					pattern: "[%d{yyyy-MM-dd hh:mm:ss} %h  %p ] %m"
				},
				backups: 10,
				//compress: true,
				encoding : 'utf-8'
			},
			data_file:{//：记录器3：输出到日期文件
				type: "dateFile",
				filename: __dirname + `/logs/${programName}`,//您要写入日志文件的路径
				alwaysIncludePattern: true,//（默认为false） - 将模式包含在当前日志文件的名称以及备份中
				daysToKeep:10,//时间文件 保存多少天，距离当前天daysToKeep以前的log将被删除
				//compress : true,//（默认为false） - 在滚动期间压缩备份文件（备份文件将具有.gz扩展名）
				layout: {
					type: "pattern",
					pattern: "[%d{yyyy-MM-dd hh:mm:ss} %h  %p ] %m"
				},
				pattern: "yyyy-MM-dd.log",//（可选，默认为.yyyy-MM-dd） - 用于确定何时滚动日志的模式。格式:.yyyy-MM-dd-hh:mm:ss.log
				encoding : 'utf-8',//default "utf-8"，文件的编码
			},
			error_file:{//：记录器4：输出到error log
				type: "dateFile",
				filename: __dirname + `/logs/${programName}_error`,//您要写入日志文件的路径
				alwaysIncludePattern: true,//（默认为false） - 将模式包含在当前日志文件的名称以及备份中
				daysToKeep:10,//时间文件 保存多少天，距离当前天daysToKeep以前的log将被删除
				//compress : true,//（默认为false） - 在滚动期间压缩备份文件（备份文件将具有.gz扩展名）
				layout: {
					type: "pattern",
					pattern: "[%d{yyyy-MM-dd hh:mm:ss} %h  %p ] %m"
				},
				pattern: "yyyy-MM-dd.log",//（可选，默认为.yyyy-MM-dd） - 用于确定何时滚动日志的模式。格式:.yyyy-MM-dd-hh:mm:ss.log
				encoding : 'utf-8',//default "utf-8"，文件的编码
				// compress: true, //是否压缩
			}
		},
		categories: {
			default:{appenders:['data_file', 'console', 'log_file'], level:'info' },//默认log类型，输出到控制台 log文件 log日期文件 且登记大于info即可
			prod:{appenders:['data_file'], level:'info'},  //生产环境 log类型 只输出到按日期命名的文件，且只输出警告以上的log
			console:{appenders:['console'], level:'debug'}, //开发环境  输出到控制台
			debug:{appenders:['console', 'log_file'], level:'debug'}, //调试环境 输出到log文件和控制台    
			error:{appenders:['error_file'], level:'error'}//error 等级log 单独输出到error文件中 任何环境的errorlog 将都以日期文件单独记录
		}
	},
	logcate: 'default'
}

// 测试环境
if (process.env.active == 'test') {
	config.plen = 2;
	config.dlen = 2;
	config.top = 555;
	config.left = 70;
	config.width = 10;
	config.serverUrl = 'http://172.27.83.63:8903/';
	config.checkchrom = 'chrome-linux/chrome --disable-background-networking';
	config.logcate = 'default';
}

// 线上环境
if (process.env.active == 'prod') {
	config.plen = 2;
	config.dlen = 5;
	config.smsTimes = 100;
	config.times = 1;
	config.top = 555;
	config.left = 70;
	config.width = 16;
	config.validWaitTimes = 6;
	config.host = 'https://m.changyoyo.com/butler/mbrauthorization.htm?';
	config.secret = 'ae4101f5ec8f4e0a';
	config.serverUrl = 'http://114.55.185.124:8070/';
	config.channelSource = '02004342';
	config.merId = 'S1000478';
	config.redis = {
		port:6379,
		db: 0,
		host:'r-bp16tofrkwh97ltewc.redis.rds.aliyuncs.com',
		pass:'mwlQy1ntBLmjGPLcN+k1qsGeLohTjVL7VexVjt1XqDb/S2XuLgPHCoiXFLXP8k7oAZ4EynHr3xQF0IJafrm7Zw=='
	};
	config.mysql = {
		host: "rm-bp1v3fzd4fc1599h7.mysql.rds.aliyuncs.com",
		port: 3306,
		user: "zx_mtk",
		password: "x3yw2lpRsrAdUQ6K0OoicEYif",
		database: "se",
		connectionLimit: 10
	};
	config.checkchrom = 'chrome-linux/chrome --disable-background-networking';
	config.logcate = 'prod';
}

module.exports = config;
